stdlib/array.mli

```diff
@@ -320,6 +320,71 @@ val of_seq : 'a Seq.t -> 'a array
 (** Create an array from the generator
     @since 4.07 *)
 
+(** {1:array_concurrency Arrays and concurrency safety}
+
+    Care must be taken when concurrently accessing arrays from multiple domains.
```

Comment:
This warning ought to be explained in the same paragraph, since documentation is often only scanned. I would add a short take-out message:
```
Care must be taken when concurrently accessing arrays from multiple domains:
accessing an array will never crash a program, but unsynchronized accesses might
yield surprising results. 
```

Reply:
> Care must be taken when concurrently accessing arrays from multiple domains: unsynchronized access to an array will never lead to crashes but may yield surprising results. 

I wonder whether it is better to use the appropriate term here -- `non-sequentially-consistent` -- instead of beating around the bush and calling it `surprising`. Sequential consistency is well understood in concurrency literature and it is not a bad thing if our users also use the terms correctly. We will need to define sequential consistency.

Reply:
Giving more context to the vague "surprising" is better but I would keep that point as a footnote:

> but may yield surprising (non-sequentially-consistent)  results.

?

Reply:
LGTM.